lib/libvfio-user.c

```diff
@@ -1963,6 +2002,10 @@ vfu_addr_to_sg(vfu_ctx_t *vfu_ctx, vfu_dma_addr_t dma_addr,
         return ERROR_INT(EINVAL);
     }
 
+#ifdef DEBUG
```

Comment:
actually, this is only a bug depending on the client: they might have exclusion separately, or just be single-threaded.

We should also check if there's a quiesce callback set.

Reply:
> We should also check if there's a quiesce callback set.

good point

> actually, this is only a bug depending on the client: they might have exclusion separately, or just be single-threaded.

I don't follow, can you elaborate?

Reply:
If the client is single threaded, then by definition, it can't be executing vfu_map_sg() or friends at the same time as handling a dma unregister. So it's safe. Also true if it happens to take its own pthread lock around those. But, presumably, if they have a quiesce cb, then neither of those are true, and it makes sense to double check them.

Reply:
Right, makes sense. Is there something you want done about this, e.g. a comment in the code?

Reply:
Yes, I'm suggesting you only assert if there's a quiesce callback registered

Reply:
oh sure